lib/rides/models/ride.dart

```diff
@@ -96,8 +96,12 @@ class Ride extends Trip {
   }
 
   static Future<Ride?> rideOfUserAtTime(DateTime start, DateTime end, int userId) async {
-    //get all rides of user
-    final List<Ride> rides = Ride.fromJsonList(await supabaseClient.from('rides').select().eq('rider_id', userId));
+    //get all approved of user
+    final List<Ride> rides = await getRidesOfUser(userId).then(
```

Comment:
why do you use "then"?

just write the call in the next line. There is an await there anyways

Reply:
changed it